pkg/issuer/vault/issue.go

```diff
@@ -49,6 +50,8 @@ const (
 	messageCertIssued = "Certificate issued successfully"
 
 	defaultCertificateDuration = time.Hour * 24 * 90
+
+	kubernetesServiceTokenPath = "/var/run/secrets/kubernetes.io/serviceaccount/token"
```

Comment:
Can we change this to be a field specified on the issuer, as a `secretRef`?

Users can then create a service account with no permissions, dedicated for authentication with Vault on a per-issuer basis.

This would be a lot more secure than handing off/using cert-manager's own service account credentials for all Vault authentication 😄

Reply:
Sure that sounds reasonable. Just keep in mind that that service account will still need a minimum permission of system:auth-delegator in order to call the TokenReview API.